Support local testing with one k8s VM.

jrajahalme opened the pull request:
Delete all old VMs regardless of the value of K8S_NODES.

Do not try to reprovision k8s2 when K8S_NODES=1.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/9681)
<!-- Reviewable:end -->

jrajahalme committed:
test: Delete all old vms before starting new ones

Delete all old VMs regardless the value of K8S_NODES.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

jrajahalme committed:
test: Support reprovisioning with K8S_NODES=1

Do not try to reprovision k8s2 when K8S_NODES=1.

This makes local k8s CI testing more feasible in situations where the
local machine does not have memory for running the two VMs. Note that
some tests are likely to fail due to resource limitiations when only
one VM is provisioned.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

ianvernon commented on test/test_suite_test.go:
```diff
@@ -242,23 +240,30 @@ var _ = BeforeAll(func() {
 		// Boot / provision VMs if specified by configuration.
 		if config.CiliumTestConfig.Reprovision {
+			var nodesInt int
+			nodes := os.Getenv(k8sNodesEnv)
```
Is there a reason this can't be a CLI argument to `ginkgo`? We have a mix of environment variables and CLI flags, which is confusing.

jrajahalme replied:
Maybe, but I didn't want to change the interface in this PR.

aanm replied:
just pinging @nebril  in case it wants to give any input...

nebril replied:
We will move all env vars to CLI flags soon, seems like it's out of scope of this PR.

jrajahalme committed:
test: Bump K8S_VERSION to 1.17 in vagrant-local-start

Bump K8S_VERSION to 1.17 in vagrant-local-start and change the default
K8S_NODES to 2.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>
